_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,693 @@
+
+In this tutorial, we'll use React since it seems to be [the most popular](https://trends.google.com/trends/explore?q=angular,react,vuejs) frontend framework nowadays. 
+
+## Get Started with JHipster 7
+
+To get started with JHipster, you'll need a fast internet connection and Node.js installed. The project recommends you use the latest LTS (Long Term Support) version, which is 14.7.6 at the time of this writing. To run the app, you'll need to have Java 11 installed. If you have Git installed, JHipster will auto-commit your project after creating it. This will allow you to upgrade between versions.
+
+Run the following command to install JHipster:
+
+```shell
+npm i -g generator-jhipster@7
+```
+
+To create a full-stack app with JHipster, create a directory, and run `jhipster` in it:
+
+```shell
+mkdir full-stack-java
+cd full-stack-java
+jhipster
+```
+
+JHipster will prompt you for the type of application to create and what technologies you'd like to include. For this tutorial, make the following choices:
+
+| Question | Answer |
+|---|---|
+| Type of application? | `Monolithic application` |
+| Name? | `flickr2` |
+| Spring WebFlux? | `No` |
+| Java package name? | `com.auth0.flickr2`  |
+| Type of authentication? | `OAuth 2.0 / OIDC` |
+| Type of database? | `SQL` |
+| Production database? | `PostgreSQL` |
+| Development database? | `H2 with disk-based persistence` |
+| Which cache? | `Ehcache` |
+| Use Hibernate 2nd level cache? | `Yes` |
+| Maven or Gradle? | `Maven` |
+| Use the JHipster Registry? | `No` |
+| Other technologies? | `<blank>` |
+| Client framework? | `React` |
+| Admin UI? | `Yes` |
+| Bootswatch theme? | `United` > `Dark` |
+| Enable i18n? | `Yes` |
+| Native language of application? | `English` |
+| Additional languages? | `Portuguese (Brazilian)` |
+| Additional testing frameworks? | `Cypress` |
+| Install other generators? | `No` |
+
+Press `Enter` and JHipster will create your app in the current directory and run `npm install` to install all the dependencies specified in `package.json`.
+
+### Verify Everything Works with Cypress and Keycloak
+
+When you choose OAuth 2.0 and OIDC for authentication, the users are stored outside of the application, rather than in it. You need to configure an identity provider (IdP) to store your users and allow your app to retrieve information about them. By default, JHipster ships with a Keycloak file for Docker Compose. A default set of users and groups is imported at startup, and it has a client registered for your JHipster app.
```

Comment:
```suggestion
When you choose OAuth 2.0 and OIDC for authentication, the users are stored outside of the application rather than in it. You need to configure an identity provider (IdP) to store your users and allow your app to retrieve information about them. By default, JHipster ships with a Keycloak file for Docker Compose. A default set of users and groups is imported at startup, and it has a client registered for your JHipster app.
```